WatchDog/Program.cs

```diff
@@ -28,15 +30,31 @@ public static void Main()
 
             var random = new Random();
 
-            while (true)
+            // in combination with the akka.cluster.role.frontend.min-nr-of-members = 1 setting,
+            // ensures that we don't start distributing work until at least 1 worker has joined
+            Cluster.Get(cluster).RegisterOnMemberUp(() =>
```

Comment:
This `Cluster.RegisterOnMemberUp` callback gets invoked when the watchdog node is marked as `Up` - but because of the `akka.cluster.role.frontend.min-nr-of-members = 1` setting, that can't happen until at least 1 worker node has joined.